pkg/cmd/repo/create/create.go

```diff
@@ -349,90 +333,324 @@ func createRun(opts *CreateOptions) error {
+// create repo on remote host from existing local repo
+func createFromLocal(opts *CreateOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	cs := opts.IO.ColorScheme()
+	isTTY := opts.IO.IsStdoutTTY()
+	stdout := opts.IO.Out
+
+	cfg, err := opts.Config()
+	if err != nil {
+		return err
+	}
+	host, err := cfg.DefaultHost()
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		opts.Source, err = interactiveSource()
+		if err != nil {
+			return err
+		}
+	}
+
+	repoPath := opts.Source
+
+	var baseRemote string
+	if opts.Remote == "" {
+		baseRemote = "origin"
+	} else {
+		baseRemote = opts.Remote
+	}
+
+	projectDir, projectDirErr := git.GetDirFromPath(repoPath)
+	if projectDirErr != nil {
+		return projectDirErr
+	}
+	if projectDir != ".git" {
+		return fmt.Errorf("path not a local repository")
+	}
+
+	abs, err := filepath.Abs(repoPath)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		opts.Name, opts.Description, opts.Visibility, err = interactiveRepoInfo(filepath.Base(abs))
+		if err != nil {
+			return err
+		}
+	}
+
+	var repoToCreate ghrepo.Interface
+
+	// repo name will be currdir name or specified
+	if opts.Name == "" {
+		repoToCreate = ghrepo.NewWithHost("", filepath.Base(abs), host)
+	} else if strings.Contains(opts.Name, "/") {
+		var err error
+		repoToCreate, err = ghrepo.FromFullName(opts.Name)
+		if err != nil {
+			return fmt.Errorf("argument error: %w", err)
+		}
+	} else {
+		repoToCreate = ghrepo.NewWithHost("", opts.Name, host)
+	}
+
+	input := repoCreateInput{
+		Name:              repoToCreate.RepoName(),
+		Visibility:        opts.Visibility,
+		OwnerLogin:        repoToCreate.RepoOwner(),
+		TeamSlug:          opts.Team,
+		Description:       opts.Description,
+		HomepageURL:       opts.Homepage,
+		HasIssuesEnabled:  !opts.DisableIssues,
+		HasWikiEnabled:    !opts.DisableWiki,
+		GitIgnoreTemplate: opts.GitIgnoreTemplate,
+		LicenseTemplate:   opts.LicenseTemplate,
+	}
+
+	repo, err := repoCreate(httpClient, repoToCreate.RepoHost(), input)
+	if err != nil {
+		return err
+	}
+
+	if isTTY {
+		fmt.Fprintf(stdout,
+			"%s Created repository %s on GitHub\n",
+			cs.SuccessIconWithColor(cs.Green),
+			ghrepo.FullName(repo))
+	}
+
+	protocol, err := cfg.Get(repo.RepoHost(), "git_protocol")
+	if err != nil {
+		return err
+	}
+
+	remoteURL := ghrepo.FormatRemoteURL(repo, protocol)
+
+	if opts.Interactive {
+		var addRemote bool
+		remoteQuesiton := &survey.Confirm{
+			Message: `Add a remote?`,
+			Default: true,
+		}
+		err = prompt.SurveyAskOne(remoteQuesiton, &addRemote)
+		if err != nil {
+			return err
+		}
+
+		if !addRemote {
+			return nil
+		}
+
+		pushQuestion := &survey.Input{
+			Message: "What should the new remote be called?",
+			Default: "origin",
+		}
+		err = prompt.SurveyAskOne(pushQuestion, &baseRemote)
+		if err != nil {
+			return err
+		}
+	}
+
+	//get the current branch
+	currentBranch, err := git.CurrentBranch()
```

Comment:
I don't think we need to query the current branch at all. If needed, it's possible to push the current branch without knowing the name of the branch:

```
git push -u origin HEAD
```

Furthermore, this queries the branch from the git repository of the current directory, but if the `--source` value was something other than `.`, then we would have to query the current branch from that directory, not from the current one.